tools/gulp/tasks/payload.ts

```diff
@@ -138,16 +142,34 @@ async function uploadPayloadResults(database: firebaseAdmin.database.Database, c
   }
 }
 
-/** Gets the last payload uploaded from previous Travis builds. */
-async function getLastPayloadResults(database: firebaseAdmin.database.Database) {
-  const snapshot = await database.ref('payloads')
-    .orderByChild('timestamp')
-    .limitToLast(1)
-    .once('value');
+/** Gets payload results of the specified commit sha. */
+async function getPayloadResults(database: firebaseAdmin.database.Database, commitSha: string) {
+  const snapshot = await database.ref('payloads').child(commitSha).once('value');
+
+  if (!snapshot.exists()) {
+    throw `There is no payload data uploaded for SHA ${commitSha}`;
+  }
 
-  // The value of the DataSnapshot is an object with the SHA as a key. Later only the
-  // first value of the object will be returned because the SHA is unnecessary.
-  const results = snapshot.val();
+  return snapshot.val();
+}
 
-  return snapshot.hasChildren() ? results[Object.keys(results)[0]] : null;
+/** Gets the SHA of the commit where the payload was uploaded before this Travis Job started. */
+function getCommitFromPreviousPayloadUpload(): string {
+  if (isTravisMasterBuild()) {
+    const commitRange = process.env['TRAVIS_COMMIT_RANGE'];
+    // In some situations Travis will include multiple commits in a single Travis Job. This means
+    // that we can't just resolve the previous commit by using the parent commit of HEAD.
+    // By resolving the amount of commits inside of the current Travis Job we can figure out
+    // how many commits before HEAD the last Travis Job ran.
+    const commitCount = spawnSync('git', ['rev-list', '--count', commitRange]).stdout
+      .toString().trim();
+    // With the amount of commits inside of the current Travis Job we can query Git to print
```

Comment:
`With the amount of commits inside of the current Travis job we can` -> `With the amount of commits inside of the current Travis job, we can`